File: the-media-live-stream/typescript/lib/the-media-live-stream-website.ts
import * as cdk from '@aws-cdk/core';
import { SPADeploy } from 'cdk-spa-deploy';
const fs = require('fs')

export class TheMediaLiveStreamWebsiteStack extends cdk.Stack {
  
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    let url = "";
    let that = this;

    // Getting URL from mediapackage
    fs.exists('./urlwebsite.json', (exists:any) => {
      if (exists) {
        var jsonFile = require('../urlwebsite.json');
        let jsondata = JSON.parse(JSON.stringify(jsonFile));
        if(jsondata.hasOwnProperty('TheMediaLiveStreamStack')){
          url = jsondata["TheMediaLiveStreamStack"]["mediapackageurlstream"];
        }

        // Writing new file
        let data = fs.readFileSync('website/index_original.html').toString('utf-8'); {
          let dataWithUrl = data.replace("##URLMEDIA##", url);
          fs.writeFile('website/index.html', dataWithUrl, function(err:any) {
            if (err) {
                return console.error(err);
            }else {
              new SPADeploy(that, "S3MediaLiveExample").createBasicSite({indexDoc: 'index.html',
                                                                          websiteFolder: 'website/'});
            }
          });
        };

      }
    });

  }
}

File: the-media-live-stream/python/the_media_live_stream/the_media_live_stream_website.py
import json
import os

from aws_cdk import (
    core
)

from spa_deploy import SPADeploy

class TheMediaLiveStreamWebsiteStack(core.Stack):

    def __init__(self, scope: core.Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        url = ""
        # Getting URL export from CfnOutput
        if os.path.isfile('urlwebsite.json') and os.access('urlwebsite.json', os.R_OK):
            with open('urlwebsite.json') as json_file:
                data = json.load(json_file)
                if "the-media-live-stream" in data:
                    url = data["the-media-live-stream"]["mediapackageurlstream"]

        with open("../python/website/index_original.html", "rt") as index_o:
            with open("../python/website/index.html", "wt") as index_f:
                for line in index_o:
                    index_f.write(line.replace('##URLMEDIA##', url))
        
        SPADeploy(scope=self, id='S3MediaLiveExample').create_basic_site(index_doc="index.html",
                                                                         website_folder="../python/website")
